### PR TITLE
Add fire lines

### PR DESCRIPTION
**Pull request by pjanik**
This PR adds complete fire line UI implementation, time limit, length limit and some bug fixes and refactoring. Commit names should be more or less descriptive.

### CODE REVIEW

File: src/components/view-3d/marker.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    const marker = getEntity();
+    if (marker) {
+      marker.material = defMaterial.current;
+    }
+    return () => {
+      defMaterial!.current!.map!.dispose();
+      defMaterial!.current!.dispose();
+      highlightMaterial!.current!.map!.dispose();
+      highlightMaterial!.current!.dispose();
+    };
```
Comment (dstrawberrygirl) [inline src/components/view-3d/marker.tsx]: Seems like a lot! of! not-null!

Comment (pjanik) [reply src/components/view-3d/marker.tsx]: Yeap, my biggest problem with hooks and refs is actually that everything can be `null` and there're so many `if`s required (or `!`).

**Comment by pjanik** [comment]
Right, I've assumed touch event will be a separate story in the future.
Re design, that's true as well, I didn't have better idea for now, and I guess Micheal can change that in the future anyway.

File: src/components/view-3d/marker.tsx (hunk 1 above)
Comment (kswenson) [reply src/components/view-3d/marker.tsx]: In TypeScript 3.7, you can use optional chaining (`?.`) for the best of both worlds -- test and proceed if present.

Comment (pjanik) [reply src/components/view-3d/marker.tsx]: Good to know, thanks!

**Pull request merged**